quic/s2n-quic-core/src/event/generated.rs

```diff
@@ -6834,6 +6834,573 @@ mod traits {
         }
     }
 }
+pub mod metrics {
+    use super::*;
+    use crate::event::metrics::Recorder;
+    #[derive(Clone, Debug)]
+    pub struct Subscriber<P: Provider> {
+        provider: P,
+    }
+    impl<P: Provider> Subscriber<P> {
+        pub fn new(provider: P) -> Self {
+            Self { provider }
+        }
+    }
+    pub struct Context<R: Recorder> {
+        recorder: R,
+        pub application_protocol_information: u32,
+        pub server_name_information: u32,
+        pub packet_skipped: u32,
+        pub packet_sent: u32,
+        pub packet_received: u32,
+        pub active_path_updated: u32,
+        pub path_created: u32,
+        pub frame_sent: u32,
+        pub frame_received: u32,
+        pub packet_lost: u32,
+        pub recovery_metrics: u32,
+        pub congestion: u32,
+        pub ack_processed: u32,
+        pub rx_ack_range_dropped: u32,
+        pub ack_range_received: u32,
+        pub ack_range_sent: u32,
+        pub packet_dropped: u32,
+        pub key_update: u32,
+        pub key_space_discarded: u32,
+        pub connection_started: u32,
+        pub connection_closed: u32,
+        pub duplicate_packet: u32,
+        pub transport_parameters_received: u32,
+        pub datagram_sent: u32,
+        pub datagram_received: u32,
+        pub datagram_dropped: u32,
+        pub connection_id_updated: u32,
+        pub ecn_state_changed: u32,
+        pub connection_migration_denied: u32,
+        pub handshake_status_updated: u32,
+        pub tls_exporter_ready: u32,
+        pub path_challenge_updated: u32,
+        pub tls_client_hello: u32,
+        pub tls_server_hello: u32,
+        pub rx_stream_progress: u32,
+        pub tx_stream_progress: u32,
+        pub keep_alive_timer_expired: u32,
+        pub mtu_updated: u32,
+        pub slow_start_exited: u32,
+        pub delivery_rate_sampled: u32,
+        pub pacing_rate_updated: u32,
+        pub bbr_state_changed: u32,
+        pub dc_state_changed: u32,
+    }
+    pub trait Provider: 'static + Send + Sync {
+        type Recorder: Recorder;
+        fn recorder(
+            &mut self,
+            meta: &api::ConnectionMeta,
+            info: &api::ConnectionInfo,
+        ) -> Self::Recorder;
+    }
```

Comment:
So thinking about this more... What if this was just a `super::Subscriber` instead? The interface is the exact same as `Subscriber`, just different names. And while we're at it, we should also delegate each of the functions to the inner subscriber so the concrete impl can set any custom fields it wants to.